internals/models.py

```diff
@@ -323,20 +323,11 @@ def del_none(d):
 class DictModel(ndb.Model):
   # def to_dict(self):
   #   return dict([(p, unicode(getattr(self, p))) for p in self.properties()])
-  saved = False  # class variable provides default value
-
-  @classmethod
-  def _post_get_hook(cls, key, future):
-    obj = future.get_result()
-    if obj is not None:
-      # test needed because post_get_hook is called even if get() fails!
-      obj.saved = True
-
-  def _post_put_hook(self, future):
-    self.saved = True
 
   def is_saved(self):
-    return self.saved
+    if self.key:
+      return True 
+    return False
 
```

Comment:
The hook `_post_get_hook` only gets called when using `Key.get()` to fetch the entries. Since, we are using `query()` at many places (e.g. to fetch the list of all features), this hook will not be helpful to us. Hence, I have come up with a work-around. 

In old DB Client Library `is_saved()` returns `True` if the model instance has been written (put) to the Datastore at least once. Also the following is mentioned [here](https://cloud.google.com/appengine/docs/standard/python/datastore/modelclass), 
> This method only checks that the instance has been written to the Datastore at least once since it was created. It does not check whether the instance's properties have been updated since the last time it was written.

Hence, we need someway to figure out if the data has been written to the datastore. Since we do not specify the key while creating an entity (except in the case of `SharedInvalidate`), the key is `None` until the entity is written (using `put()` ). Hence, checking if `key` is not `None` could mean that the entity has been written to the datastore.

Reply:
Yes, that sounds like it would work.